pkg/pillar/cmd/zedrouter/appcontainer.go

```diff
@@ -139,3 +153,94 @@ func appStatsMayNeedReinstallACL(ctx *zedrouterContext, config types.AppNetworkC
 		}
 	}
 }
+
+func getAppContainerLogs(status types.AppNetworkStatus, last map[string]string) {
+	var buf bytes.Buffer
+	cli, containers, err := getAppContainers(status)
+	if err != nil {
+		return
+	}
+
+	for _, container := range containers {
+		var lasttime, newtime, message string
+		containerName := strings.Trim(container.Names[0], "/")
+		if lt, ok := last[containerName]; ok {
+			lasttime = lt
+		}
+		out, err := cli.ContainerLogs(context.Background(), container.ID, apitypes.ContainerLogsOptions{
+			ShowStdout: true,
+			ShowStderr: true,
+			Timestamps: true,
+			Since:      lasttime,
+		})
+		if err != nil {
+			log.Errorf("getAppContainerLogs: log output error %v", err)
+			continue
+		}
+
+		stdcopy.StdCopy(&buf, &buf, io.LimitReader(out, 100000))
+		logLines := strings.Split(buf.String(), "\n")
+		log.Debugf("getAppContainerLogs: container %s, lasttime %s, lines %d", containerName, lasttime, len(logLines))
+		for _, line := range logLines {
+			sline := strings.SplitN(line, " ", 2)
+			time := sline[0]
```

Comment:
What is the format of the timestamp? I wonder if it would be clearer and easier if this was first parsed to time.Time and then compare and store it.

Reply:
I don't think it's a RFC3339 format. but I would think the API will understand RFC3339. I will try that to convert.